### DISCUSSION TITLE
Plans for v2.0.0

### DISCUSSION THREAD

**Discussion by bekoenig** [bekoenig/getdown#1] [Announcements]
After years of a minimal life support getdown will be [revived under new ownership](https://github.com/threerings/getdown/pull/260#issuecomment-1793594646).

The first stable [release of version 2.0.0](https://github.com/bekoenig/getdown/tree/release/2.0.0) will be the starting point for further contributions. This release will contain a lot of improvements:
- removed deprecated parent pom
- inlined samskivert utils code for swing
- minimized custom utils
- increased compile level to java 8
- necessary dependency updates 
- consistent logging over slf4j without system outs
- modernized code format
- dropped compression and obfuscation using proguard 
- release management using github actions
- new group-id and packages

Every big milestone will be published as snapshot to sonatype.

The [old repository](https://github.com/threerings/getdown) will be marked as deprecated and not longer supported after the final release. The wiki and some side project for getdown will move soon too. 

Getdown is a great project, which should be keep alive! Feel free to contribute and discus the next steps.

**Comment by samskivert** [comment]
Great, thanks so much for taking over the project!

**Reply by bekoenig** [reply]
After some struggles with implementation details the first milestone is reached: The [snapshots are available at sonatype](https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/bekoenig/getdown/) and contain all advertised and some more improvements.  

The next days I want to setup the github actions and perform the first alpha-release.

**Reply by bekoenig** [reply]
Hej @serge-xav and @samskivert,

the first release candidate is available [now](https://central.sonatype.com/namespace/io.github.bekoenig.getdown)!

**Reply by serge-xav** [reply]
Hi @bekoenig I encountered 2issues with the 2.0.0-SNAPSHOT : 

1. the getdown-core.jar can't be used standalone ; I got exception below. The move to slf4J seems the cause
 `Exception in thread "main" java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
	at io.github.bekoenig.getdown.tools.Digester.<clinit>(Digester.java:34)
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 1 more`

2. the getdown-launcher has a dependency to jregistrykey » jregistrykey which is not available on Maven

**Reply by JohReher** [reply]
Hi @serge-xav,
thanks for testing out the new 2.0.0-SNAPSHOT!

Regarding the jregistrykey:
The jregistrykey 1.0 is bundled in the lib directory and imported in maven as local repository. You are right that's not found on maven central or other default maven repositories. Maybe there is a problem with your checked out directory structure, so that the relative pathes don't match?

**Reply by serge-xav** [reply]
In the threerings/getdown, the jregistrykey was added in the POM as optional dependency, which is no more the case here.
So when I'm trying to add getdown-launcher as a dependency of my own project, I get dependency error, and was obliged to add  exclude("jregistrykey","jregistrykey") in my gradle/kotlin project configuration

**Reply by bekoenig** [reply]
Thank you @serge-xav for your reply:

1. You are right that core is no longer a standalone jar after shift to slf4j. If you want to use tools from core please use the new published artifact with classifier `jar-with-dependencies`. This note is now included in the [migration guide](https://github.com/bekoenig/getdown/wiki/Migrating-from-1.8-to-2.0). One of the next steps will be a refactoring to apache-httpclient to support redirects, resume resource downloads, different auth mechanisms and so on. So inclusion of third party libs in core will be the new standard pattern.
2. I removed optional, because the proxy feature is not working under windows without this dependency. So I thought it is 
unexpected to use optional when this feature does not work. But you are right that this change results in a compile error. I added optional again.

**Comment by bekoenig** [comment]
Hej @samskivert,

the final release is completed now!

The [changelog](https://github.com/bekoenig/getdown/blob/master/CHANGELOG.md) contains all improvements. A [migration-guide](https://github.com/bekoenig/getdown/wiki/Migrating-from-1.8-to-2.0) is available too.

I would be glad, if you link my project! You can mark your project as deprecated.

Greetings
